src/server/VRxControl.py

```diff
@@ -0,0 +1,339 @@
+
+    def registerController(self, controller):
+        if hasattr(controller, 'name'):
+            if controller.name in self.controllers:
+                logger.warning('Overwriting VRx Controller "{0}"'.format(controller.name))
+
+            controller.manager = self
+            self.controllers[controller.name] = controller
+            logger.info('Importing VRx Controller {}'.format(controller.name))
+        else:
+            logger.warning('Invalid controller')
+
+    def isEnabled(self):
+        if not self.enabled:
+            return False 
+
+        for controller in self.controllers.values(): # should this check be done?
+            if controller.ready:
+                return True
+
+        return False
+
+    def kill(self):
+        logger.info('Killing VRx Control')
+        self.enabled = False
+        return True
+
+    def updateStatus(self):
+        for controller in self.controllers.values():
+            controller.updateStatus()
+
+    def getControllerStatus(self):
+        status = {}
+        for controller in self.controllers.values():
+            status[controller.name] = controller.getStatus()
+        return status
+
+    def getAllDeviceStatus(self):
+        devices = {}
+        for controller in self.controllers.values():
+            status = controller.getAllDeviceStatus()
+            for device in status:
+                manager_device_id = str(controller.name) + ":" + str(device.id)
+                devices[manager_device_id] = device
+
+        return devices
+
+    def getDevices(self):
+        devices = {}
+        for controller in self.controllers.values():
+            for device in controller.devices:
+                manager_device_id = str(controller.name) + ":" + str(device.id)
+                devices[manager_device_id] = device
+
+        return devices
+
+    def getActiveDevices(self, seat, pilot_id):
+        devices = {}
+        for controller in self.controllers.values():
+            for device in controller.devices:
+                manager_device_id = str(controller.name) + ":" + str(device.id)
+                if device.method == VRxDeviceMethod.ALL:
+                    devices[manager_device_id] = device
+                elif device.method == VRxDeviceMethod.PILOT and device.pilot_id == pilot_id:
+                    devices[manager_device_id] = device
+                elif device.method == VRxDeviceMethod.SEAT and device.pilot_id == seat:
+                    devices[manager_device_id] = device
+
+        return devices
+
+    def setDeviceMethod(self, device_id, method):
+        for controller in self.controllers.values():
+            for device in controller.devices:
+                manager_device_id = str(controller.name) + ":" + str(device.id)
+                if manager_device_id == device_id:
+                    controller.setDeviceMethod(device.id, method)
+                    return True
+        return False
```

Comment:
I don't see anything reacting to these results. And return-values to communicate "didn't do anything" are usually not the way to go in Python. If it is important to check for the result of this, use an exception (and catch it of course).